src/Controller/Component/CsrfComponent.php

```diff
@@ -93,7 +93,7 @@ public function startup(Event $event)
         if ($request->is('get') && $cookieData === null) {
             $this->_setCookie($request, $response);
         }
-        if ($request->is(['put', 'post', 'delete', 'patch']) || !empty($request->data)) {
+        if ($request->is(['put', 'post', 'delete', 'patch']) || !empty($request->data())) {
```

Comment:
Isn't the `` !empty()`` part now obsolete, as the ``data()`` method prevents notices?